custom_components/midea_ac_lan/config_flow.py

```diff
@@ -353,9 +357,11 @@ async def async_step_auto(
                             format((PRESET_ACCOUNT[0] ^ PRESET_ACCOUNT[2]), "X"),
                         ).decode("ASCII"),
                     )
-                    if not await self.cloud.login():
+                    if self.cloud and not await self.cloud.login():
```

Comment:
same not necessary

Reply:
Did you copy this PR locally, change the code and tru mypy ?